pkgs/development/python-modules/numpy/default.nix

```diff
@@ -54,6 +56,23 @@ in buildPythonPackage rec {
     # Patching of numpy.distutils is needed to prevent it from undoing the
     # patch to distutils.
     ./numpy-distutils-C++.patch
+    # f2py.f90mod_rules generates code with invalid function pointer conversions, which are
+    # clang 16 makes an error by default.
+    (fetchpatch {
+      url = "https://github.com/numpy/numpy/commit/609fee4324f3521d81a3454f5fcc33abb0d3761e.patch";
+      hash = "sha256-6Dbmf/RWvQJPTIjvchVaywHGcKCsgap/0wAp5WswuCo=";
+    })
+  ] ++ lib.optionals (lib.versionOlder version "1.25") [
```

Comment:
We should be able to sleep name this unconditional

Reply:
I can make this patch apply unconditionally. It can be dropped with the derivation is updated to numpy 1.25.